### PR TITLE
fix for issue #88

### PR DESCRIPTION
**Pull request by andyDoucette**
A fix for issue #88 .

Here i use the /private/BalanceEx endpoint to get the balance and corresponding amount held by a open order for each nonzero currency balance.  That allows me to very easily calculate get_balances in a way that accepts both "USD" and "ZUSD"

For my use case, I would probably directly use get_balances_and_excluded().  I assume that is what "BalanceEx" means.

If you're looking for info on BalanceEx, the only mention I've found of it from Kraken is in their [suggested API wrapper](https://support.kraken.com/hc/en-us/articles/360025180232-REST-API-command-line-client-Python-).  Their kind support people also suggested it to me.

### CODE REVIEW

File: kraken/spot/user/__init__.py [hunk 1]
```diff
@@ -112,23 +126,15 @@ def get_balances(self: "User", currency: str) -> dict:
...
         balance: Decimal = Decimal(0)
+        available_balance: Decimal(0)
+        
         curr_opts: tuple = (currency, f"Z{currency}", f"X{currency}")
-        for symbol, value in self.get_account_balance().items():
+        for symbol, data in self.get_balances_and_excluded().items():
             if symbol in curr_opts:
-                balance = Decimal(value)
+                balance = Decimal(data['balance'])
+                available_balance=balance-Decimal(data['hold_trade'])
```
Comment (btschwertfeger) [inline kraken/spot/user/__init__.py:135]:
```suggestion
                available_balance=balance-Decimal(data["hold_trade"])
```

File: kraken/spot/user/__init__.py [hunk 2]
```diff
@@ -112,23 +126,15 @@ def get_balances(self: "User", currency: str) -> dict:
...
 
         balance: Decimal = Decimal(0)
+        available_balance: Decimal(0)
+        
         curr_opts: tuple = (currency, f"Z{currency}", f"X{currency}")
-        for symbol, value in self.get_account_balance().items():
+        for symbol, data in self.get_balances_and_excluded().items():
             if symbol in curr_opts:
-                balance = Decimal(value)
+                balance = Decimal(data['balance'])
```
Comment (btschwertfeger) [inline kraken/spot/user/__init__.py:134]:
```suggestion
                balance = Decimal(data["balance"])
```

File: kraken/spot/user/__init__.py [hunk 3]
```diff
@@ -112,23 +126,15 @@ def get_balances(self: "User", currency: str) -> dict:
         """
 
         balance: Decimal = Decimal(0)
+        available_balance: Decimal(0)
```
Comment (btschwertfeger) [inline kraken/spot/user/__init__.py:129]:
```suggestion
        available_balance: Decimal = Decimal(0)
```

File: kraken/spot/user/__init__.py [hunk 4]
```diff
@@ -84,6 +84,20 @@ def get_account_balance(self: "User") -> dict:
             method="POST", uri="/private/Balance"
         )
 
+    def get_balances_and_excluded(self: "User") -> dict:
+        """
+        Returns the currenzies with a nonzero balance and the corresponding amount held by an open trade.
```
Comment (btschwertfeger) [inline kraken/spot/user/__init__.py:89]:
```suggestion
        Returns the currencies with a non-zero balance and the corresponding amount held by open orders.
```

**Review by btschwertfeger — CHANGES_REQUESTED** [review]

File: kraken/spot/user/__init__.py [hunk 5]
```diff
@@ -84,6 +84,20 @@ def get_account_balance(self: "User") -> dict:
...
+         ``balance`` (inclding value in orders), and ``hold_trade``
+         (amount is in orders)
+        :rtype: dict
+        """
+        return self._request(  # type: ignore[return-value]
+            method="POST", uri="/private/BalanceEx"
+        )
+
+
     def get_balances(self: "User", currency: str) -> dict:
```
Comment (btschwertfeger) [inline kraken/spot/user/__init__.py:101]: What about:

```suggestion
    def get_balance(self: "User", currency: str) -> dict:
```

File: kraken/spot/user/__init__.py [hunk 6]
```diff
@@ -84,6 +84,20 @@ def get_account_balance(self: "User") -> dict:
             method="POST", uri="/private/Balance"
         )
 
+    def get_balances_and_excluded(self: "User") -> dict:
```
Comment (btschwertfeger) [inline kraken/spot/user/__init__.py:87]: The naming might be better, even its a breaking change.

```suggestion
    def get_balances(self: "User") -> dict:
```

**Review by btschwertfeger — CHANGES_REQUESTED** [review]

**Comment by btschwertfeger** [comment]
Hello @andyDoucette, thank you for your contribution!

I'm interested in merging your changes after you addressed the following issues: 
* Please add a unit test for the new method/endpoint. 
* Make sure that the CI/CD workflow run through and all jobs are green.
* Also see the suggestions I added above.

Have a nice day!

**Comment by andyDoucette** [comment]
Thanks for your feedback.  I wasn't expecting to have to do more work to share the work I have already done with the rest of the world.  I'll try to get to this sometime, but to be honest I have a lot of other things vying for my time, so I am not sure when I would get to this.  If you are ok polishing this up, feel free.  If I get a break sometime in the next few months, maybe I will, but I would put it at a 20% chance or less.

I just feel good that at least the work I did already to solve the problem for myself is "out there" in case it makes anyone else's work a little easier.

**Comment by andyDoucette** [comment]
I agree that the new naming is better, if you don't mind breaking changes or don't have many users yet.

**Comment by btschwertfeger** [comment]
> Thanks for your feedback. I wasn't expecting to have to do more work to share the work I have already done with the rest of the world. I'll try to get to this sometime, but to be honest I have a lot of other things vying for my time, so I am not sure when I would get to this. If you are ok polishing this up, feel free. If I get a break sometime in the next few months, maybe I will, but I would put it at a 20% chance or less.
> 
> I just feel good that at least the work I did already to solve the problem for myself is "out there" in case it makes anyone else's work a little easier.

So I will do the last steps if that's ok for you.

**Comment by andyDoucette** [comment]
That would be amazing.  Thank you so much for making this library, btw!

**Comment by btschwertfeger** [comment]
This enhancement will be addressed in the separate PR #98.

**Pull request closed without merging**